test/integration/scheduler/volume_binding_test.go

```diff
@@ -361,23 +474,110 @@ func TestPVAffinityConflict(t *testing.T) {
 	}
 }
 
-func setupCluster(t *testing.T, nsName string, numberOfNodes int) *testConfig {
-	// Enable feature gates
-	utilfeature.DefaultFeatureGate.Set("VolumeScheduling=true,PersistentLocalVolumes=true")
+// TestRescheduleProvisioning tests:
+// 1. If scheduler will trigger provisioning by adding "selected-node" annotation
+// 2. If PV controller will reschedule provisioning that failed to provision
+// by removing the annotation.
+// To clearly verify the annotation is added/removed, we need to start/stop
```

Comment:
I'm not sure if this will be any simpler, but instead of starting/stopping controllers, could you implement the FakeProvisioner to keep a global counter.  Fail provisioning the first time, and then succeed the 2nd time?  Also the FakeProvisioner could create a PV with the selected node name in the path, so you could also validate the correct node was passed in.

Reply:
Main purpose here is to validate the provisioning reschedule behavior here.
And if we don't set breakpoint to check the annotation detail, we can hardly tell that  controller reschedule provisioning as expected.
Even though we get a failure count and a successful creation, there is still a chance that the controller does nothing to the annotation, it just run into the sync loop twice with the same annotation.